executorlib/standalone/hdf.py

```
@@ -15,21 +15,22 @@
     """
-    with h5py.File(file_name, "a") as fname:
-        for data_key, data_value in data_dict.items():
-            if data_key in group_dict.keys():
-                fname.create_dataset(
-                    name="/" + group_dict[data_key],
-                    data=np.void(cloudpickle.dumps(data_value)),
-                )
+    if file_name is not None:
+        with h5py.File(file_name, "a") as fname:
+            for data_key, data_value in data_dict.items():
+                if data_key in group_dict.keys():
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)